test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -845,7 +886,14 @@ private void CompareSelectedR4VectorColumns(string leftColumnName, string rightC
 
                     Assert.Equal(expected.Length, actual.Length);
                     for (int i = 0; i < expected.Length; ++i)
-                        Assert.Equal(expected.GetItemOrDefault(i), actual.GetItemOrDefault(i), precision);
+                    {
+                        // We are using float values. But the Assert.Equal function takes doubles.
+                        // And sometimes the converted doubles are different in their precision.
+                        // So make sure we compare floats
+                        float exp = expected.GetItemOrDefault(i);
+                        float act = actual.GetItemOrDefault(i);
+                        Assert.True((Math.Abs(exp - act) < prec) || (float.IsNaN(exp) && float.IsNaN(act)));
```

Comment:
There’s an overload to assert.equals that takes a precision. You don’t need to do this yourself.

Reply:
That is the problem (as described in the comments). The precision comparison happens after the conversion to double. And there seems to be some bug in the conversion to double where the precision isn't preserved during the conversion.

Reply:
In that case, then check out:

https://github.com/dotnet/machinelearning/blob/fabdabf867dc68481d519740a96b981bbeb63180/test/Microsoft.ML.TestFramework/BaseTestBaseline.cs#L572-L574

Which is how we handle this case for the rest of the tests. The advantage of using that method is that it will log the values of each in the failure. Where your test code above will just say "Failed expected `true` but got `false`", which doesn't help when investigating a failure.

Reply:
Resolved with latest checkin.